### PR TITLE
[Feature] support instance_aware_colorization inference

### PR DESCRIPTION
**Pull request by zengyh1900**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Support instance_aware_colorization inference mentioned in https://github.com/open-mmlab/mmediting/issues/1247 

## Modification

- [x] built upon  https://github.com/open-mmlab/mmediting/pull/1173
- [x] use mmdet instead of detectron 
- [x] add docstring 
- [x] add uni tests 
- [x] update requirements 
- [x] update readme and model zoo 

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

**Before PR**:

- [x] I have read and followed the workflow indicated in the [CONTRIBUTING.md](https://github.com/open-mmlab/mmediting/blob/master/.github/CONTRIBUTING.md) to create this PR.
- [x] Pre-commit or linting tools indicated in [CONTRIBUTING.md](https://github.com/open-mmlab/mmediting/blob/master/.github/CONTRIBUTING.md) are used to fix the potential lint issues.
- [x] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [x] New functionalities are covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [x] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [x] If the modification has potential influence on downstream or other related projects, this PR should be tested with some of those projects.
- [x] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

**Comment by zengyh1900** [comment]
woman_gray
woman_color
child_gray
child_color
giraffe_gray
giraffe_color

File: mmedit/apis/colorization_inference.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import torch
+from mmengine.dataset import Compose
+from mmengine.dataset.utils import default_collate as collate
+from torch.nn.parallel import scatter
+
+
+def colorization_inference(model, img):
+    """Inference image with the model.
```
Comment (LeoXing1996) [inline mmedit/apis/colorization_inference.py:9]: Type lint in the function signature is missing. (Maybe can convert to a **good first issue**)

File: mmedit/datasets/transforms/crop.py [hunk 2]
```diff
@@ -916,3 +921,107 @@ def __repr__(self):
...
+                torch.long)
+            results['box_info_8x'] = torch.from_numpy(box_info_8x).type(
+                torch.long)
+            results['empty_box'] = False
+        else:
+            results['empty_box'] = True
+        return results
+
+    def predict_bbox(self, image):
+        lab_image = cv.cvtColor(image, cv.COLOR_BGR2LAB)
```
Comment (LeoXing1996) [inline mmedit/datasets/transforms/crop.py:1014]: Type lint in the function signature is missing. (Maybe can convert to a **good first issue**)

File: mmedit/models/editors/inst_colorization/color_utils.py [hunk 3]
```diff
@@ -0,0 +1,307 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import numpy as np
+import torch
+
+
+def xyz2rgb(xyz):
+    """Conversion images from lab to xyz.
```
Comment (LeoXing1996) [inline mmedit/models/editors/inst_colorization/color_utils.py:7]: Type lint in the function signature is missing. (Maybe can convert to a **good first issue**)

Comment (LeoXing1996) [inline mmedit/models/editors/inst_colorization/color_utils.py:7]: Conversion images from lab to xyz. ->   Conversion images from xyz to rgb.

File: mmedit/models/editors/inst_colorization/color_utils.py [hunk 4]
```diff
@@ -0,0 +1,307 @@
+# Copyright (c) OpenMMLab. All rights reserved.
```
Comment (LeoXing1996) [inline mmedit/models/editors/inst_colorization/color_utils.py:1]: Maybe we can move colorspace transformation functions in this file to MMCV later. (Maybe can convert to a **good first issue**)

File: mmedit/models/editors/inst_colorization/colorization_net.py [hunk 5]
```diff
@@ -0,0 +1,313 @@
...
+        classification (bool): backprop trunk using classification,
+            otherwise use regression. Default: True
+    """
+
+    def __init__(self,
+                 input_nc,
+                 output_nc,
+                 norm_type,
+                 use_tanh=True,
+                 classification=True):
```
Comment (LeoXing1996) [inline mmedit/models/editors/inst_colorization/colorization_net.py:35]: Type lint in the function signature is missing. (Maybe can convert to a **good first issue**)

Comment (zengyh1900) [reply mmedit/models/editors/inst_colorization/colorization_net.py:35]: Agree

File: mmedit/models/editors/inst_colorization/color_utils.py (hunk 4 above)
Comment (zengyh1900) [reply mmedit/models/editors/inst_colorization/color_utils.py:1]: Agree.

File: mmedit/models/editors/inst_colorization/color_utils.py (hunk 3 above)
Comment (zengyh1900) [reply mmedit/models/editors/inst_colorization/color_utils.py:7]: Agree.

File: mmedit/datasets/transforms/crop.py (hunk 2 above)
Comment (zengyh1900) [reply mmedit/datasets/transforms/crop.py:1014]: Agree.

**Review by LeoXing1996 — APPROVED** [review]

**Pull request merged**